components/sidebar/index.tsx

```diff
@@ -1,59 +1,247 @@
+	timeFrame: TimeFrame;
+}): PersonalStatsProps => {
+	const walletStats = wallets[wallet] ? wallets[wallet][timeFrame] : undefined;
 
-const DesktopItems = styled.div`
-	display: none;
+	return {
+		rank: walletStats?.rankPosition ?? 0,
+		dataUploaded: walletStats?.byteCount ?? 0,
+		filesUploaded: walletStats?.fileCount ?? 0,
+		daysStreaked: 0,
+	};
+};
 
-	@media ${device.tablet} {
-		display: grid;
-		gap: 1rem;
-	}
-`;
+type GroupStatsProps = {
+	uploaders: number;
+	dataUploaded: number;
+	filesUploaded: number;
+	streakers: number;
+};
+const getGroupStats = ({ wallets, timeFrame }: { wallets: Wallets; timeFrame: TimeFrame }): GroupStatsProps => {
+	const walletValues = Object.values(wallets);
 
-const Sidebar = ({
+	return walletValues.reduce(
+		(accumulator, stats) => {
+			const { byteCount, fileCount } = stats[timeFrame];
+			const didUploaded = byteCount > 0;
+
+			return {
+				dataUploaded: accumulator.dataUploaded + byteCount,
+				filesUploaded: accumulator.filesUploaded + fileCount,
+				uploaders: didUploaded ? accumulator.uploaders + 1 : accumulator.uploaders,
+				streakers: accumulator.streakers,
+			};
+		},
+		{
+			dataUploaded: 0,
+			filesUploaded: 0,
+			uploaders: 0,
+			streakers: 0,
+		},
+	);
+};
+
+const GroupStats = ({
 	uploaders,
-	data,
-	files,
+	dataUploaded,
+	filesUploaded,
 	streakers,
 }: {
-	uploaders?: number;
-	data?: number;
-	files?: number;
-	streakers?: number;
+	uploaders: number;
+	dataUploaded: number;
+	filesUploaded: number;
+	streakers: number;
 }): JSX.Element => (
-	<Wrapper>
-		<MobileItems>
-			<Countdown />
-			{uploaders && <Box text={uploaders} description='Uploaders' />}
-		</MobileItems>
-		<DesktopItems>
-			{data && <Box text={`${data} GB`} description='Data Uploaded' />}
-			{files && <Box text={files} description='Files Uploaded' />}
-			{streakers && <Box text={streakers} description='Streakers' arrowUp />}
-		</DesktopItems>
-	</Wrapper>
+	<>
+		<Box text={uploaders} description='Uploaders' />
+		<Box text={formatBytes(dataUploaded)} description='Data Uploaded' />
+		<Box text={filesUploaded} description='Files Uploaded' />
+		<Box text={streakers} description='Streakers' />
+	</>
 );
 
+const PersonalStats = ({
+	rank,
+	dataUploaded,
+	filesUploaded,
+	streaksCompleted,
+	daysStreaked,
+	isMobile,
+}: {
+	rank: number;
+	dataUploaded: number;
+	filesUploaded?: number;
+	streaksCompleted?: number;
+	daysStreaked?: number;
+	isMobile: boolean;
+}): JSX.Element => (
+	<>
+		<Box text={rank} description='Rank' />
+		{daysStreaked !== undefined && <Box text={daysStreaked} description='Days Streaked' />}
+		<Box text={formatBytes(dataUploaded)} description={isMobile ? 'Your Data' : 'Data Uploaded'} />
+		{filesUploaded !== undefined && <Box text={filesUploaded} description='Files Uploaded' />}
+		{streaksCompleted !== undefined && <Box text={streaksCompleted} description='Streaks Completed' />}
+	</>
+);
+
+const Sidebar = ({ data }: { data: Data }): JSX.Element => {
```

Comment:
This component has gained a considerable amount of responsibilities here

Reply:
a lot, but I was not sure how to break it

Reply:
I think we could tackle this one together, but given the timeframe and scope it is better for another time. We can revisit this if we ever expand this app for other reward programs

Reply:
I agree